app/modules/node.c

```diff
@@ -120,24 +120,87 @@ static int node_sleep( lua_State* L )
 #endif //PMSLEEP_ENABLE
 static int node_info( lua_State* L )
 {
-  lua_pushinteger(L, NODE_VERSION_MAJOR);
-  lua_pushinteger(L, NODE_VERSION_MINOR);
-  lua_pushinteger(L, NODE_VERSION_REVISION);
-  lua_pushinteger(L, system_get_chip_id());   // chip id
-  lua_pushinteger(L, spi_flash_get_id());     // flash id
-  lua_pushinteger(L, flash_rom_get_size_byte() / 1024);  // flash size in KB
-  lua_pushinteger(L, flash_rom_get_mode());
-  lua_pushinteger(L, flash_rom_get_speed());
-  lua_pushstring(L, BUILDINFO_BRANCH);
-  lua_pushstring(L, BUILDINFO_COMMIT_ID);
-  lua_pushstring(L, BUILDINFO_RELEASE);
-  lua_pushstring(L, BUILDINFO_RELEASE_DTS);
-  lua_pushboolean(L, BUILDINFO_SSL);
-  lua_pushstring(L, BUILDINFO_LFS);
-  lua_pushstring(L, BUILDINFO_MODULES);
-  lua_pushstring(L, BUILDINFO_BUILD_TYPE);
-  
-  return 16;
+  const char* options[] = {"hw", "sw_version", "build_config", "legacy", NULL};
+  int option = luaL_checkoption (L, 1, options[3], options);
+
+  switch (option) {
+    case 0: { // hw
+      lua_createtable (L, 0, 5);
+      int table_index = lua_gettop(L);
+      lua_pushliteral(L, "chip_id");
```

Comment:
You would normally use `lua_setfield()` here:
```C
      lua_pushinteger(L, system_get_chip_id());   // chip id
      lua_setfield(L, table_index,  "chip_id");
```
or even define a composite `setintfield(L,ndx,field,i)` and ditto for  `setstrfield(L,ndx,field,s)`:
```C
      setintfield(L, table_index,  "chip_id", system_get_chip_id());
```

Reply:
This seems like something that several modules want; at least my `sntppkt` wants it too.  How'd you feel about extending the lua utility library in the (our?) `luaN_` namespace?

Reply:
Using lua_setfield now.
It seems somewhat contradictionary to first split the information in several parts because they are too many and then introduce an "all" option to have it all again.
So I would like to not implement that unless there is a strong demand for it.